Chat cleanup

justinbmeyer opened the pull request:
This adds `New APIs Used` notes after every code example.  It also fixes the tests (and example code) for when the generators are fixed: https://github.com/donejs/generator-donejs/issues/214

chasenlehara committed:
Fix the issue links in the contribution guide

Some of the links were to the wrong projects.

justinbmeyer committed:
adds New APIs Used sections, fixes tests to avoid *

justinbmeyer committed:
fixed callback descriptions

phillipskevin commented on docs/guides/guide.md:
```diff
@@ -76,7 +82,7 @@ Go to [http://localhost:8080/](localhost:8080) to see our application showing a
 ## Adding Bootstrap
 
-DoneJS makes it easy to import other projects that are published on [NPM](https://npmjs.org). In this section we will install and add [Bootstrap](http://getbootstrap.com/) to the page and see DoneJS's [hot module swapping](./Features.html#hot-module-swapping) in action.
+DoneJS makes it easy to import other projects that are published on [NPM](https://npmjs.org). In this section, we will install and add [Bootstrap](http://getbootstrap.com/) to the page, and see DoneJS's [hot module swapping](./Features.html#hot-module-swapping) in action.
```
I don't think the second comma is needed since the list only has two items
- install and add bootstrap to the page
- see hot module swapping in action

chasenlehara replied:
If the concern was having two “and”s in the sentence, this could be rephrased as “In this section, we will install [Bootstrap](http://getbootstrap.com/), add it to the page, and see DoneJS's [hot module swapping](./Features.html#hot-module-swapping) in action.”

matthewp committed:
Use the http registry

If this works it will fix #864.

Some Appveyor VMs have an expired certificate for npmjs.org, and when
this occurs it causes the install to fail. This change makes it so that
Appveyor uses the http registry instead.
